src/main/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkComponent.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.serialization;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Representation of a CloudSdkComponent. Used for json serialzation/deserialization.
+ */
+public class CloudSdkComponent {
```

Comment:
Because if new components are added, we will be able to deserialize them into this class without making any code changes. Even if they don't change that often, I really like having the guarantee that parsing won't break when they do.

You can certainly catch JsonParseExceptions, but it would still be better to be able to just return an actual object right?